pyqtgraph/palette.py

```diff
@@ -0,0 +1,72 @@
+from .Qt import QtGui
+
+
+def dark_QPalette():
+    RED = QtGui.QColor('#F00000')
+    BLACK = QtGui.QColor('#000000')
+    BG_DARK = QtGui.QColor('#19232D')
+    BG_NORMAL = QtGui.QColor('#37414F')
+    BG_LIGHT = QtGui.QColor('#455364')
+    FG_DARK = QtGui.QColor('#9DA9B5')
+    FG_NORMAL = QtGui.QColor('#E0E1E3')
+    FG_LIGHT = QtGui.QColor('#F0F0F0')
+    SEL_DARK = QtGui.QColor('#1A72BB')
+    SEL_NORMAL = QtGui.QColor('#26486B')
+    SEL_LIGHT = QtGui.QColor('#346792')
+    qpal = QtGui.QPalette(QtGui.QColor(BG_DARK))
+
+    for ptype in (QtGui.QPalette.Active, QtGui.QPalette.Inactive):
+        qpal.setColor(ptype, QtGui.QPalette.Base, BG_DARK)
+        qpal.setColor(ptype, QtGui.QPalette.Window, BG_DARK)
+        qpal.setColor(ptype, QtGui.QPalette.WindowText, FG_NORMAL)
+        qpal.setColor(ptype, QtGui.QPalette.AlternateBase, BG_LIGHT)
+        qpal.setColor(ptype, QtGui.QPalette.Button, BG_LIGHT)
+        qpal.setColor(ptype, QtGui.QPalette.ButtonText, FG_LIGHT)
+        qpal.setColor(ptype, QtGui.QPalette.Highlight, SEL_LIGHT)
+        qpal.setColor(ptype, QtGui.QPalette.HighlightedText, FG_LIGHT)
+        qpal.setColor(ptype, QtGui.QPalette.Text, FG_LIGHT)
+        qpal.setColor(ptype, QtGui.QPalette.ToolTipBase, BG_LIGHT)
+        qpal.setColor(ptype, QtGui.QPalette.ToolTipText, FG_LIGHT)
+    qpal.setColor(QtGui.QPalette.Disabled, QtGui.QPalette.Base, BG_NORMAL)
+    qpal.setColor(QtGui.QPalette.Disabled, QtGui.QPalette.Button, BG_NORMAL)
+    qpal.setColor(QtGui.QPalette.Disabled, QtGui.QPalette.ButtonText, FG_DARK)
+    qpal.setColor(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, FG_DARK)
+    qpal.setColor(QtGui.QPalette.Disabled, QtGui.QPalette.Text, FG_DARK)
+    qpal.setColor(QtGui.QPalette.Disabled, QtGui.QPalette.Highlight, BG_LIGHT)
+    qpal.setColor(QtGui.QPalette.Disabled, QtGui.QPalette.HighlightedText, FG_DARK)
+    return qpal
+
+
+def light_QPalette():
```

Comment:
Since we are keeping these private, should we give them more specific names?
`getQDarkStyleDarkQPalette()` and `getQDarkStyleLightQPalette()`, maybe?
I think we might want to keep the namespace open for our own version of just "light" and "dark" palettes. 
The very dark colors in QDarkStyle do not work well with e.g. black outlines of UI elements and the black shading applied according to button state. It really needs the stylesheet.
I'd ultimately like to have a dark mode palette that works without depending on an external stylesheet, and I would want to refer to that as just "dark". :)

Reply:
I'm all for `QDarkStyleDarkQPalette()` and `QDarkStyleLightQPalette()` but pep8 doesn't like the java style getters/setter names. We might even want to make these properties.

> I'd ultimately like to have a dark mode palette that works without depending on an external stylesheet, and I would want to refer to that as just "dark".

Agree!  [more about this here](https://github.com/pyqtgraph/pyqtgraph/pull/2101/files/4a565507d760cc8cd917134a32d322e9832786e2#r755812152)

Reply:
I think content of palette.py needs some more thought.
This might better use some internal dictionary that is easy to add to. Ideally, the user should be able to throw in their own palette info into the colors folder and grab it from there in multiple applications.

At this time, the strategy might be to keep everything in here private and marked experimental. `applyPalette` is the only supported access to this... for now.

Reply:
Regarding the naming: 
We are basically following the Qt CamelCase naming conventions, at least on the API side. So we regularly deviate from python standards. @j9ac9k has the best overview here, though.

Reply:
@Wubbzi we generally deviate from pep8 when it comes to naming convention to public portions of our API as @NilsNemitz suggested.  We will go `snake_case` for methods intended for internal methods, and use the leading underscore to designate if something should be considered private or not.

On top of using `camelCase` for public API methods; we try and follow Qt naming conventions when feasible (for example we do this with `.setData()` `.getData()` and a variety of other methods that we feel are analogous to other parts of the Qt API.

Of course the above are just guidelines.  ... regarding `applyPalette` would it be more appropriate to use `.setPalette()`?  That method is used in other parts of the Qt framework: 
![image](https://user-images.githubusercontent.com/646398/143372020-553a0f35-6757-4669-8fbc-332c91210b4c.png)

Reply:
> We are basically following the Qt CamelCase naming conventions, at least on the API side

I don't have any problem with the case conventions pg uses. The snake_case/true_property stuff that was added in pyside6 looks so wrong, and I hope it never catches on. 

I was trying to stick with the Qt conventions. for example, Qt has `setMaximumHeight()` but when you want to get it, its just `maximumHeight()` not `getMaximumHeight()` I stuck with the capital Q because I was afraid it would look weird. Qt does lowercase their methods that begin with q, like `qAbs`,`qRound` but you don't see those used in Python very often.

Reply:
@j9ac9k oooh, another dash user!

> We will go snake_case for methods intended for internal methods, and use the leading underscore to designate if something should be considered private or not.

Oh, I didn't realize thats what was meant by 
> the strategy might be to keep everything in here private and marked experimental. 

> regarding applyPalette would it be more appropriate to use .setPalette()
Yep. I was trying to stick as closely to what Nils suggested as I could, so after I copied it out of chat I never changed it, but it has now been changed.

Reply:
Well, my thought process wasn't that sophisticated.
More like "make sure nobody uses it, whatever we put now, we'll break it later when we put more functionality there"

Reply:
> another dash user!

if you're on macOS and a developer and _not_ using Alfred + Dash ...that's a crime 😆